contracts/mainnet/connectors/euler/main.sol

```diff
@@ -0,0 +1,552 @@
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function borrow(
+		uint256 subAccount,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		bool isEth = token == ethAddr ? true : false;
+		address _token = isEth ? wethAddr : token;
+
+		IEulerDToken borrowedDToken = IEulerDToken(
+			markets.underlyingToDToken(_token)
+		);
+		borrowedDToken.borrow(subAccount, amt);
+
+		convertWethToEth(isEth, TokenInterface(_token), _amt);
+
+		setUint(setId, _amt);
+
+		_eventName = "LogBorrow(uint256,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(subAccount, token, _amt, getId, setId);
+	}
+
+	/**
+	 * @dev Repay ETH/ERC20_Token.
+	 * @notice Repay a token from Euler
+	 * @param subAccount Subaccount number
+	 * @param token The address of the token to repay.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of the token to repay. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function repay(
+		uint256 subAccount,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		bool isEth = token == ethAddr ? true : false;
+
+		address _token = isEth ? wethAddr : token;
+		IEulerDToken borrowedDToken = IEulerDToken(
+			markets.underlyingToDToken(_token)
+		);
+
+		if (isEth) convertEthToWeth(isEth, TokenInterface(_token), _amt);
+
+		_amt = _amt == type(uint256).max
+			? borrowedDToken.balanceOf(address(this))
+			: _amt;
+
+		TokenInterface(_token).approve(EULER_MAINNET, _amt);
+		borrowedDToken.repay(subAccount, amt);
+
+		setUint(setId, _amt);
+
+		_eventName = "LogRepay(uint256,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(subAccount, token, _amt, getId, setId);
+	}
+
+	/**
+	 * @dev Mint ETH/ERC20_Token.
+	 * @notice Mint a token from Euler. Maint creates equal amount of deposits and debts.
+	 * @param subAccount Subaccount number
+	 * @param token The address of the token to mint.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of the token to mint.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function mint(
+		uint256 subAccount,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		bool isEth = token == ethAddr ? true : false;
+		address _token = isEth ? wethAddr : token;
+		IEulerEToken eToken = IEulerEToken(markets.underlyingToEToken(_token));
+
+		if (isEth) convertEthToWeth(isEth, TokenInterface(_token), _amt);
```

Comment:
Is mint amount `uint256(-1)` possible?

Reply:
`mint` is like leverage. So I believe, no. Although there can be a workaround, if it's `uint256(-1)`, then we calculate the safe Max for the position and send that value. I think we can skip it for now.